apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardEngineImpl.java

```diff
@@ -267,52 +284,57 @@ public MigrateState relinkDataToSnapshotBlock(CommandParamInfo paramInfo) {
-            log.debug("Trimmed '{}' to HEIGHT '{}' within {} sec", "PUBLIC_KEY", paramInfo.getSnapshotBlockHeight(), (System.currentTimeMillis() - startTrim) / 1000);
+//            trimService.doTrimDerivedTables(paramInfo.getSnapshotBlockHeight().intValue(), sourceDataSource); // TRIM 'PUBLIC_KEY' table before processing
+            log.debug("Trimmed '{}' to HEIGHT '{}' within {} sec", "PUBLIC_KEY", paramInfo.getSnapshotBlockHeight(), (System.currentTimeMillis() - startTrim)/1000);
 
             for (String tableName : paramInfo.getTableNameList()) {
                 long start = System.currentTimeMillis();
                 currentTable = tableName;
 
-                BatchedPaginationOperation paginationOperationHelper = helperFactory.createSelectInsertHelper(tableName);
-                processOneTableByHelper(paramInfo, sourceConnect, tableName, start, paginationOperationHelper);
+                Optional<BatchedPaginationOperation> paginationOperationHelper = helperFactory.createSelectInsertHelper(tableName, true);
+                if (paginationOperationHelper.isPresent()) {
+                    processOneTableByHelper(paramInfo, sourceConnect, tableName, start, paginationOperationHelper);
+                    paginationOperationHelper.get().reset();
+                } else {
+                    log.warn("NO processing HELPER class for table '{}'", tableName);
+                }
                 recovery = updateShardRecoveryProcessedTableList(sourceConnect, currentTable, DATA_RELINK_STARTED);
             }
             state = DATA_RELINKED_IN_MAIN;
             updateToFinalStepState(sourceConnect, recovery, state);
             sourceConnect.commit();
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("Error RELINK processing table = '" + currentTable + "'", e);
             sourceDataSource.rollback(false);
             state = MigrateState.FAILED;
             return state;
-        }
-        finally {
+        } finally {
             if (sourceDataSource != null) {
                 sourceDataSource.commit();
             }
         }
-        log.debug("RELINK Processed table(s)=[{}] in {} secs", paramInfo.getTableNameList().size(), (System.currentTimeMillis() - startAllTables) / 1000);
+        log.debug("RELINK Processed table(s)=[{}] in {} secs", paramInfo.getTableNameList().size(), (System.currentTimeMillis() - startAllTables)/1000);
         return state;
     }
+*/
 
     private void processOneTableByHelper(CommandParamInfo paramInfo, Connection sourceConnect,
                                          String tableName, long start,
-                                         BatchedPaginationOperation batchedPaginationOperation) throws Exception {
+                                         Optional<BatchedPaginationOperation> paginationOperationHelper) throws Exception {
```

Comment:
Do not use optional as parameter: https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments/31923105

Reply:
Done